test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -2447,73 +2451,94 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
         [Fact]
         public void ConnectionClosedWhenResponseDoesNotSatisfyMinimumDataRate()
         {
-            var chunkSize = 64 * 1024;
-            var chunks = 128;
-            var responseSize = chunks * chunkSize;
+            using (StartLog(out var loggerFactory))
+            {
+                var logger = loggerFactory.CreateLogger($"{typeof(ResponseTests).FullName}.{nameof(ConnectionClosedWhenResponseDoesNotSatisfyMinimumDataRate)}");
 
-            var requestAborted = new ManualResetEventSlim();
-            var messageLogged = new ManualResetEventSlim();
-            var mockKestrelTrace = new Mock<IKestrelTrace>();
-            mockKestrelTrace
-                .Setup(trace => trace.ResponseMininumDataRateNotSatisfied(It.IsAny<string>(), It.IsAny<string>()))
-                .Callback(() => messageLogged.Set());
+                var chunkSize = 64 * 1024;
+                var chunks = 128;
+                var responseSize = chunks * chunkSize;
 
-            var testContext = new TestServiceContext
-            {
-                Log = mockKestrelTrace.Object,
-                SystemClock = new SystemClock(),
-                ServerOptions =
+                var requestAborted = new ManualResetEventSlim();
+                var messageLogged = new ManualResetEventSlim();
+                var mockKestrelTrace = new Mock<KestrelTrace>(loggerFactory.CreateLogger("Microsoft.AspNetCore.Server.Kestrel")) { CallBase = true };
```

Comment:
:+1: I see you found `CallBase`. I don't think you can change this on a per-method basis though.

Reply:
I had CallBase there before, but it requires that the method be virtual, hence the whole virtual/non-virtual discussion above :). I actually don't want it per-method. I want it on for all methods including ones I add Callbacks to (i.e. run the base logic after the callback or something), but Moq doesn't support that, so for that one method that I actually mock out, the base behavior won't run.